uvicorn/config.py

```diff
@@ -300,22 +313,27 @@ def load(self):
             encoded_headers
             if b"server" in dict(encoded_headers)
             else [(b"server", b"uvicorn")] + encoded_headers
-        )  # type: List[Tuple[bytes, bytes]]
+        )
 
         if isinstance(self.http, str):
-            self.http_protocol_class = import_from_string(HTTP_PROTOCOLS[self.http])
+            http_protocol_class = import_from_string(HTTP_PROTOCOLS[self.http])
+            self.http_protocol_class: Type[asyncio.Protocol] = http_protocol_class
         else:
             self.http_protocol_class = self.http
 
         if isinstance(self.ws, str):
-            self.ws_protocol_class = import_from_string(WS_PROTOCOLS[self.ws])
+            ws_protocol_class = import_from_string(WS_PROTOCOLS[self.ws])
+            self.ws_protocol_class: Optional[Type[asyncio.Protocol]] = ws_protocol_class
         else:
             self.ws_protocol_class = self.ws
 
         self.lifespan_class = import_from_string(LIFESPAN[self.lifespan])
 
         try:
-            self.loaded_app = import_from_string(self.app)
+            if isinstance(self.app, str):
+                self.loaded_app = import_from_string(self.app)
+            else:
+                self.loaded_app = self.app
```

Comment:
why do we do that ? afair it's done already in the import_from_string def, isn't it ?

Reply:
Yeah, good point, we can just let `import_from_str` handle this. Reverted.